utils.groovy

```diff
@@ -282,4 +282,110 @@ def get_artifacts_to_build(pipecfg, stream, basearch) {
+def upload_to_clouds(pipecfg, basearch, buildID, stream) {
+    // Get artifacts to upload
+    def meta_json = "builds/${buildID}/${basearch}/meta.json"
+    def meta = readJSON file: meta_json
+    def artifacts = meta['images'].keySet() as List
+    def clouds = pipecfg.clouds.keySet() as List
+    // Check if we have cloud config to upload these artifacts
+    def uploads = artifacts.intersect(clouds)
+    parallel uploads.inject([:]) { branch, cloud -> branch["Upload ${cloud}"] = {
+
+        def var_name = cloud.toUpperCase()
+        tryWithCredentials([file(variable: "${var_name}_BUILD_UPLOAD_CONFIG",
+                                 credentialsId: "${cloud}-build-upload-config")]) {
+
+            def bucket = pipecfg.clouds."${cloud}".bucket
+            def path = pipecfg.clouds."${cloud}".path
+
+            if (cloud in ['aws', 'aliyun']) {
+                def primary_region = pipecfg.clouds."${cloud}".primary_region
+                def grant_user_args
+                if (cloud == 'aws') {
+                    def aws_accounts = pipecfg.clouds."${cloud}".upload_accounts
+                    grant_user_args = aws_accounts.collect{"--grant-user ${it}"}.join(" ")
+                    bucket = "s3://${bucket}/$path"
+                }
+                shwrap("""
+                cosa buildextend-${cloud} \
+                    --upload \
+                    --arch=${basearch} \
+                    --build=${buildID} \
+                    --region=${primary_region} ${grant_user_args} \
+                    --bucket ${bucket} \
+                    --credentials-file=\${${var_name}_BUILD_UPLOAD_CONFIG}
+                """)
+           }
+           if (cloud == 'azure') {
+               def container = pipecfg.clouds."${cloud}".container
+               def resource_group = pipecfg.clouds."${cloud}".resource_group
+               def storage_account = pipecfg.clouds."${cloud}".storage_account
+               shwrap("""cosa buildextend-azure \
+                   --upload \
+                   --auth \${AZURE_IMAGE_UPLOAD_CONFIG} \
+                   --build=${buildID} \
+                   --container ${container} \
+                   --profile \${AZURE_IMAGE_UPLOAD_CONFIG} \
+                   --resource-group ${resource_group} \
+                   --storage-account ${storage_account} \
+                   --force
+                """)
+           }
+           if (cloud == 'gcp') {
+                def create_image =  (pipecfg.clouds."${cloud}".create_image ? "--create-image=true": "")
+                def deprecated =  (pipecfg.clouds."${cloud}".deprecated ? "--deprecated": "")
+                def description =  (pipecfg.clouds."${cloud}".description ? ('--description=' + pipecfg.clouds."${cloud}".description): "")
+                def family =  (pipecfg.clouds."${cloud}".family ? '--family ' + pipecfg.clouds."${cloud}".family: "")
+                def acl = (pipecfg.clouds."${cloud}".public ? "--public" : "")
```

Comment:
I don't think we pass this for FCOS but I think we don't need to because the project is configured to be public. Does RHCOS pass `--public` when uploading to GCP?

Reply:
Yes, it is  from `rhcos`